opentracing/context.go

```diff
@@ -0,0 +1,35 @@
+package opentracing
+
+// A ContextID is the smallest amount of state needed to describe a span's
+// context within a larger [potentially distributed] trace. The ContextID is
+// not intended to encode the span's operation name, timing, or log data, but
+// merely any unique identifiers (etc) needed to contextualize it within a
+// larger trace tree.
+//
+// ContextIDs are sufficient to propagate the, well, *context* of a particular
+// trace from process to process.
+type ContextID interface {
+	// Create a child context for this ContextID, and return both that child's
+	// own ContextID as well as any Tags that should be added to the child's
+	// Span.
+	//
+	// The returned ContextID type must be the same as the type of the
+	// ContextID implementation itself.
+	NewChild() (childCtx ContextID, initialChildSpanTags Tags)
+
+	// Serializes the ContextID as an arbitrary byte string.
+	Serialize() []byte
```

Comment:
I would suggest string, not byte, serialization by default, with other optional interfaces for other serialization formats, like bytes, or Zipkin-style 4-numbers tuple. Otherwise, the bytes need to be serialized again into a text protocol, e.g. HTTP headers, meaning the common implementation would need to do something like base64, and that makes the context ID on the wire unnecessarily opaque.